core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +77,13 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
```

Comment:
I see you have backcompat below for when this setting is passed in through a string field. However, i think we also need to have it for custom analyzers here.